api/envoy/extensions/filters/http/ext_authz/v3/ext_authz.proto

```diff
@@ -253,6 +254,16 @@ message AuthorizationResponse {
   // (Host)* will be in the response to the client. When a header is included in this list, *Path*,
   // *Status*, *Content-Length*, *WWWAuthenticate* and *Location* are automatically added.
   type.matcher.v3.ListStringMatcher allowed_client_headers = 2;
+
+  // When this :ref:`list <envoy_api_msg_type.matcher.v3.ListStringMatcher>`. is set, authorization
+  // response headers that have a correspondent match will be emitted as dynamic metadata, whose keys
+  // correspond to the name of the matched header and whose values are the respective values. When this
+  // list is *not* set, no additional dynamic metadata will be emitted. This metadata lives in a namespace
+  // specified by the canonical name of extension filter that requires it:
+  //
+  // - :ref:`envoy.filters.http.ext_authz <config_http_filters_ext_authz_dynamic_metadata>` for HTTP filter.
+  // - :ref:`envoy.filters.network.ext_authz <config_network_filters_ext_authz_dynamic_metadata>` for network filter.
+  type.matcher.v3.ListStringMatcher dynamic_metadata_from_headers = 5;
```

Comment:
This is 5 because I originally based this PR from https://github.com/envoyproxy/envoy/pull/14514 which uses index 4.